### PR TITLE
Add spi pooling and mfrc522 detection

### LINKED ISSUE #7: Add device tree with MFRC522

### LINKED ISSUE #8: Add working MFRC522 in device tree without spidev conflict

### PR DESCRIPTION
**Pull request by Skallwar**
Fixes #8
Fixes #7

### CODE REVIEW

File: module/mfrc522_module.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    pr_info("[MFRC522] SPI Probed\n");
+
+    if (client->max_speed_hz > MFRC522_SPI_MAX_CLOCK_SPEED) {
+	    pr_info("[MFRC522] Current speed is to high %u. Setting speed to %u\n", client->max_speed_hz, MFRC522_SPI_MAX_CLOCK_SPEED);
+	    client->max_speed_hz = MFRC522_SPI_MAX_CLOCK_SPEED;
+    }
+
+    return mfrc522_detect(client);
+}
+
```
Comment (CohenArthur) [inline module/mfrc522_module.c:63]: We should probably move this into `mfrc522_spi.c`

File: module/mfrc522_module.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    .probe = mfrc522_spi_probe,
+};
+
+static int mfrc522_detect(struct spi_device *client) {
+    struct address_byte version_reg_read = address_byte_build(MFRC522_SPI_READ, MFRC522_VERSION_REG);
+    char version;
+
+    spi_write(client, &version_reg_read, 1);
+    mdelay(100);
```
Comment (CohenArthur) [inline module/mfrc522_module.c:36]: Is the delay absolutely necessary?

File: module/mfrc522_module.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+MODULE_LICENSE("GPL v2");
+MODULE_AUTHOR("ks0n");
+MODULE_DESCRIPTION("Driver for the MFRC522 RFID Chip");
+
+#define MFRC522_VERSION_BASE 0x90
+#define MFRC522_VERSION_1 0x91
+#define MFRC522_VERSION_2 0x92
+#define MFRC522_VERSION_NUM(ver) ((ver) - MFRC522_VERSION_BASE)
+
+
```
Comment (CohenArthur) [inline module/mfrc522_module.c:20]:
```suggestion

```
Extra newline

File: module/mfrc522_module.c [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+static int mfrc522_detect(struct spi_device *client) {
+    struct address_byte version_reg_read = address_byte_build(MFRC522_SPI_READ, MFRC522_VERSION_REG);
+    char version;
+
+    spi_write(client, &version_reg_read, 1);
+    mdelay(100);
+    spi_read(client, &version, 1);
+
+    switch (version) {
+            case MFRC522_VERSION_1 || MFRC522_VERSION_2:
```
Comment (n1tram1) [inline module/mfrc522_module.c]: this is not rust. case expressions are constants (compile-time) this is probably not doing the intended behaviour

File: module/mfrc522_module.c (hunk 2 above)
Comment (Skallwar) [reply module/mfrc522_module.c]: I don't know

Comment (Skallwar) [reply module/mfrc522_module.c]: No f50b119

**Review by n1tram1 — CHANGES_REQUESTED** [review]

File: module/mfrc522_module.c (hunk 4 above)
Comment (Skallwar) [reply module/mfrc522_module.c]: Good catch ! Fixed by (f50b119)

File: module/mfrc522_module.c (hunk 1 above)
Comment (Skallwar) [reply module/mfrc522_module.c]: Not lines 15 to 18 and ``mfrc522_detect``, but the rest I agree

Comment (Skallwar) [reply module/mfrc522_module.c]: @n1tram1 and I think we should just wait for a regmap pull request

**Review by CohenArthur — APPROVED** [review]
LGTM :fire: I agree that we should use the regmap API and that will probably shake the architecture a bit, so we'll get back to this later

**Review by n1tram1 — APPROVED** [review]

**Pull request merged**